### PR TITLE
target: fix flash, ram size and identifier for KW41Z512xxx4

### PR DESCRIPTION
**Pull request by jfischer-phytec-iot**
The size of the Flash and RAM is wrong for target MKW41Z512xxx4,
fix that and use kw41z512xxx4 identifier for this target.

### CODE REVIEW

**Comment by jfischer-phytec-iot** [comment]
@MaureenHelm FYI

**Comment by c1728p9** [comment]
CC @mmahadevan108

**Comment by flit** [comment]
It's true the flash and RAM sizes are wrong. Looks like they were copied from KW40 and not updated.

Why do you want to change the target name to 'kw41z512xxx4'? This would be unlike all the other targets in pyOCD.

**Comment by jfischer-phytec-iot** [comment]
>Why do you want to change the target name to 'kw41z512xxx4'? This would be unlike all the other targets in pyOCD.


Regarding the Flash and RAM configuration, there are two variants of the SoC, one with 256KB Flash / 64KB RAM and 512KB Flash / 128KB RAM. Probably `kw41z512` is also clear enough.

File: pyOCD/target/__init__.py [hunk 1]
```diff
@@ -78,7 +78,7 @@
           'kv11z7': target_MKV11Z128xxx7.KV11Z7,
           'kw01z4': target_MKW01Z128xxx4.KW01Z4,
           'kw40z4': target_MKW40Z160xxx4.KW40Z4,
-          'kw41z4': target_MKW41Z512xxx4.KW41Z4,
+          'kw41z512xxx4': target_MKW41Z512xxx4.KW41Z4,
```
Comment (c1728p9) [inline pyOCD/target/__init__.py]: The name should be kept the same for now. If there is a switch to a new naming scheme, all the devices will be updated at the same time.

**Comment by flit** [comment]
We tend to directly support only the maximally-sized device variant, rather than attempting to include configurations for each. The target will work for all variants with smaller memory sizes, which is why the target name doesn't specify the memory size. This lets us support more devices with fewer targets.

When we eventually switch to CMSIS-Pack based configurations, we will be able to support all device variants directly.

**Comment by jfischer-phytec-iot** [comment]
ok, will rework and change the name back to 'kw41z4'

**Comment by flit** [comment]
Closing this PR because the fixes have been incorporated into #336.

**Pull request closed without merging**